src/openms/source/QC/Ms2IdentificationRate.cpp

```diff
@@ -149,4 +116,69 @@ namespace OpenMS
       meta.custom[meta.custom.size()] = ms2_ir;
     }
   }
+  Size Ms2IdentificationRate::getMS2Count_(const MSExperiment& exp)
+  {
+    if (exp.empty())
+    {
+      throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "MSExperiment is empty");
+    }
+
+    Size ms2_counter{};
+    for (auto const& spec : exp.getSpectra())
+    {
+      if (spec.getMSLevel() == 2)
+      {
+        ++ms2_counter;
+      }
+    }
+
+    if (ms2_counter == 0)
+    {
+      throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No MS2 spectra found");
+    }
+
+    return ms2_counter;
+  }
+
+  bool Ms2IdentificationRate::isTargetPeptide_(const PeptideIdentification& id, bool all_targets)
+  {
+    if (id.getHits().empty())
+    {
+      return false;
+    }
+    if (all_targets)
+    {
+      return true;
+    }
+    if (!(id.getHits()[0].metaValueExists("target_decoy")))
+    {
+      throw Exception::Precondition(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No target/decoy annotaion found. If you want to continue regardless use -MS2_id_rate:force_no_pep_index");
```

Comment:
`annotation`

Reply:
can you rename that `MS2_id_rate:force_no_pep_index` to `MS2_id_rate:assume_all_target` or so?